lib/tpm2_errata.h

```diff
@@ -0,0 +1,53 @@
+//**********************************************************************;
+// Copyright (c) 2017, Alibaba Group
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation
+// and/or other materials provided with the distribution.
+//
+// 3. Neither the name of Intel Corporation nor the names of its contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+#ifndef SRC_ERRATA_H
+#define SRC_ERRATA_H
+
+#include <sapi/tpm20.h>
+#include <stdbool.h>
+
+/*
+ * Errata location:
+ *   spec version + the section number of errata documented in the errata spec.
+ *
+ * Note that it is unnecessary to describe errata version because the
+ * section number should be kept consistent accoss all errata versions
+ * for a specific spec version.
+ */
+enum {
+    SPEC_116_ERRATA_2_7,
+};
+
+void tpm2_errata_init(TSS2_SYS_CONTEXT *sapi_ctx);
+bool tpm2_errata_fixup(int nr, ...);
```

Comment:
No one uses this?

Reply:
Also, you define enum's for this but pass int. Use the actual enum type where callers should pass enums, it makes it much simpler to understand your interface.

Reply:
I will submit a patch which shows how to use tpm2_errata_fixup(). The top comment gives a preview of that patch.